query/src/sessions/query_ctx.rs

```diff
@@ -424,6 +430,30 @@ impl QueryContext {
         })?;
         Ok(FunctionContext { tz })
     }
+
+    pub fn block_on_meta<F>(&self, future: F) -> Result<F::Output>
+    where
+        F: Future + Send + 'static,
+        F::Output: Send + 'static,
+    {
+        let rt = self.get_meta_runtime();
+        let (tx, rx): (
+            std::sync::mpsc::Sender<F::Output>,
+            std::sync::mpsc::Receiver<F::Output>,
+        ) = std::sync::mpsc::channel();
+
+        rt.try_spawn(async move {
+            let res = future.await;
+            tx.send(res).unwrap();
+        })?;
+        match rx.recv() {
+            Ok(v) => Ok(v),
+            Err(cause) => Err(ErrorCode::LogicalError(format!(
+                "Logical error, receive error. {:?}",
+                cause
+            ))),
+        }
+    }
```

Comment:
One concern: this method can not be called in a tokio-runtime, right? This method itself can not provide any guarantee on this. It's totally up to the caller.

Except this, everything else looks great to me:DDD

Reply:
I think it can, because `QueryNormalizer::normalize` is inside tokio-runtime.